cmd/bucket-replication.go

```diff
@@ -1322,11 +1326,13 @@ func (ri ReplicateObjectInfo) replicateAll(ctx context.Context, objectAPI Object
 	versioned := globalBucketVersioningSys.PrefixEnabled(bucket, object)
 	versionSuspended := globalBucketVersioningSys.PrefixSuspended(bucket, object)
 
-	gr, err := objectAPI.GetObjectNInfo(ctx, bucket, object, nil, http.Header{}, ObjectOptions{
-		VersionID:        ri.VersionID,
-		Versioned:        versioned,
-		VersionSuspended: versionSuspended,
-	})
+	gr, err := objectAPI.GetObjectNInfo(ctx, bucket, object, nil, http.Header{},
+		ObjectOptions{
+			VersionID:        ri.VersionID,
+			Versioned:        versioned,
+			VersionSuspended: versionSuspended,
+			NoDecryption:     true,
```

Comment:
NoDecryption should be true only for SSE_C , peer sites need not share the same KMS, so SSE-S3 and SSE-KMS encrypted objects need to be decrypted before replication.

Reply:
in case of AWS S3 are you allowed to setup two different KMS encryption for replicated buckets? - curious to know.

Reply:
in [AWS S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-config-for-kms-objects.html), the replication config has additional entries to specify the KMS key ID of destination -  the answer to your question is yes. 

Our configuration is simpler with the caveat that the KMS key ID used be identical.

Reply:
Yeah I was thinking that why not we send encrypted streams to destination regardless of sse being used.

We would be able to support something like that now. But I guess just doing this for SSE-C is okay.

Reply:
ok, for the time being I will set for SSE-C.